python/unitytrainers/meta_curriculum.py

```diff
@@ -0,0 +1,85 @@
+"""Contains the MetaCurriculum class."""
+
+import os
+from unitytrainers import Curriculum
+
+class MetaCurriculum(object):
+    """A MetaCurriculum holds curriculums. Each curriculum is associated to a particular
+    brain in the environment.
+    """
+
+    def __init__(self, curriculum_folder, default_reset_parameters):
+        """Initializes a MetaCurriculum object.
+
+        Args:
+            curriculum_folder (str): The relative or absolute path of the
+                folder which holds the curriculums for this environment.
+                The folder should contain JSON files whose names are the
+                brains that the curriculums belong to.
+            default_reset_parameters (dict): The default reset parameters
+                of the environment.
+        """
+        if curriculum_folder is None:
+            self._brains_to_curriculums = None
+        else:
+            self._brains_to_curriculums = {}
+            for curriculum_filename in os.listdir(curriculum_folder):
+                brain_name = curriculum_filename.split('.')[0]
+                curriculum_filepath = \
+                    os.path.join(curriculum_folder, curriculum_filename)
+                self._brains_to_curriculums[brain_name] = \
+                    Curriculum(curriculum_filepath, default_reset_parameters)
+
+    @property
+    def brains_to_curriculums(self):
+        """A dict from brain_name to the brain's curriculum."""
+        return self._brains_to_curriculums
+
+    @property
+    def lesson_nums(self):
+        """A dict from brain name to the brain's curriculum's lesson number."""
+        lesson_nums = {}
+        for brain_name, curriculum in self.brains_to_curriculums.items():
+            lesson_nums[brain_name] = curriculum.lesson_num
+
+        return lesson_nums
+
+    @lesson_nums.setter
+    def lesson_nums(self, lesson_nums):
+        for brain_name, lesson in lesson_nums.items():
+            self.brains_to_curriculums[brain_name].lesson_num = lesson
+
+    def increment_lessons(self, progresses):
+        """Increments all the lessons of all the curriculums in this MetaCurriculum.
+
+        Args:
+            progresses (dict): A dict of brain name to progress.
+        """
+        for brain_name, progress in progresses.items():
+            self.brains_to_curriculums[brain_name].increment_lesson(progress)
+
+
+    def set_all_curriculums_to_lesson_num(self, lesson_num):
+        """Sets all the curriculums in this meta curriculum to a specified lesson number.
+
+        Args:
+            lesson_num (int): The lesson number which all the curriculums will
+                be set to.
+        """
+        for _, curriculum in self.brains_to_curriculums.items():
+            curriculum.lesson_num = lesson_num
+
+
+    def get_config(self):
```

Comment:
I see how this can be complicated if multiple curriculums need to access the same reset parameters. Maybe in addition to display a warning, there should be an ordering of the curriculums : The lowest lesson numbers have priority or something like that.

Reply:
I'm not sure if that last part would be true. The ordering of the brain_name keys in the dict should be random so we really don't know which curriculum would take precedence.

Reply:
What I meant is that we can we can scan through the keys of the dictionary intelligently rather than using the default dictionary order. We __make__ the precedence happen.

Reply:
I'm inclined to avoid that because of the inefficiency. However, I do see the utility of adding this---it makes the lowest lesson number, the one that needs the most help, the one that controls the parameter. Let me see if I can find a way to do this without the inefficiency.